dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+                        b.control_volume.properties_in[t].\
+                            cp_mol_phase[p]
+                        / (b.control_volume.properties_in[t].\
+                           cp_mol_phase[p]
+                           - gas_constant)
+                        )
+                else:
+                    return 1
+
+        # Add constraint to determine the tank final temperature (T out)
+        # This constraint is derived from the 1st law of thermodynamics and 
+        # Ideal gas law assuming an adiabatic process
+        # Reference: Yang (2009). Int. J. of Hydrogen Energy 34, 6712-6721.
+
+        # TO DO: add pressure out equation for an isothermal process
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Dimensionles time")
+        def dim_less_time(b, t):
+            return (
+                self.control_volume.properties_in[t].flow_mol
+                * b.dt[t]
+                / sum(b.initial_material_holdup[t, p, j]
+                      for (p, j) in pc_set)
+                )
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Tank Temperature calculations")
+        def tank_temperature_calculation(b, t):
+            return b.control_volume.properties_out[t].temperature == (
+                  b.gama[t] *
+                  b.control_volume.properties_in[t].temperature
+                  + ((b.initial_state[t].temperature
+                      - b.gama[t] *
+                      b.control_volume.properties_in[t].temperature)
+                     / (1 + b.dim_less_time[t])))
+
+    def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET,
+                   solver='ipopt', optarg={'tol': 1e-6}):
+        '''
+        compressed hydrogen gas tank initialization routine.
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                           package(s) for the control_volume of the model to
+                           provide an initial state for initialization
+                           (see documentation of the specific property package)
+                           (default = None).
+            outlvl : sets output level of initialisation routine
+
+                     * 0 = no output (default)
+                     * 1 = return solver state for each step in routine
+                     * 2 = return solver state for each step in subroutines
+                     * 3 = include solver output infomation (tee=True)
+
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        '''
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        opt = pyo.SolverFactory(solver)
```

Comment:
If using the latest IDAES, you should replace this and the line below with `opt = get_solver(solver, optarg)`.